packages/zoe/src/contracts/publicAuction.js

```diff
@@ -19,131 +21,106 @@ import '../../exported';
  * price rule is followed, so the highest bidder pays the amount bid by the
  * second highest bidder.
  *
- * makeInstance() specifies the issuers and terms ({ numBidsAllowed }) specify
- * the number of bids required. An invitation for the seller is returned. The
+ * startInstance() specifies the issuers and terms ({ numBidsAllowed }) specify
+ * the number of bids required. An invitation for the seller is
+ * returned as the creatorInvitation. The
  * seller's offer should look like
  * { give: { Asset: asset }, want: { Ask: minimumBidAmount } }
  * The asset can be non-fungible, but the Ask amount should be of a fungible
  * brand.
- * The bidder invitations are available from await E(publicAPI).makeInvites(n). Each
+ * The bidder invitations can be made by calling makeBidInvitation on the creatorFacet. Each
  * bidder can submit an offer: { give: { Bid: null } want: { Asset: null } }.
  *
- * publicAPI also has methods to find out what's being auctioned
- * (getAuctionedAssetsAmounts()), or the minimum bid (getMinimumBid()).
- *
- * @param {ContractFacet} zcf
+ * @type {ContractStartFn}
  */
-const makeContract = zcf => {
-  const { rejectOffer, satisfies, assertKeywords, checkHook } = makeZoeHelpers(
-    zcf,
-  );
-
-  let {
-    terms: { numBidsAllowed },
-  } = zcf.getInstanceRecord();
-  numBidsAllowed = Nat(numBidsAllowed !== undefined ? numBidsAllowed : 3);
-
-  let sellerOfferHandle;
+const start = zcf => {
+  const {
+    maths: { Asset: assetMath, Ask: bidMath },
+  } = zcf.getTerms();
+  let { numBidsAllowed = 3 } = zcf.getTerms();
+  numBidsAllowed = Nat(numBidsAllowed);
+
+  let sellSeat;
   let minimumBid;
   let auctionedAssets;
-  const allBidHandles = [];
+  const bidSeats = [];
 
   // seller will use 'Asset' and 'Ask'. buyer will use 'Asset' and 'Bid'
-  assertKeywords(harden(['Asset', 'Ask']));
+  assertIssuerKeywords(zcf, harden(['Asset', 'Ask']));
 
-  const bidderOfferHook = offerHandle => {
+  /** @type {OfferHandler} */
+  const bid = bidSeat => {
     // Check that the item is still up for auction
-    if (!zcf.isOfferActive(sellerOfferHandle)) {
+    if (sellSeat.hasExited()) {
       const rejectMsg = `The item up for auction is not available or the auction has completed`;
-      throw rejectOffer(offerHandle, rejectMsg);
+      throw bidSeat.kickOut(rejectMsg);
     }
-    if (allBidHandles.length >= numBidsAllowed) {
-      throw rejectOffer(offerHandle, `No further bids allowed.`);
+    if (bidSeats.length >= numBidsAllowed) {
```

Comment:
These checks are redundant since the sellSeat will be exited synchronously when there are enough bids.